cpp/include/legate_dataframe/core/allocator.hpp

```diff
@@ -104,34 +104,5 @@ class TaskMemoryResource : public DeviceMemoryResource {
   std::unordered_map<const void*, MemAlloc> buffers_{};
 };
 
-/**
- * @brief Global RMM resource that return device memory allocations
- *
- * This resource returns allocations backed by either legate buffers or regular CUDA
- * buffers depending on whether the allocation is called from inside a legate task
- * or by the control code. Therefore, this resource can be used both inside and
- * outside tasks and can be used with RMM's `set_per_device_resource()` and
- * `set_current_device_resource()`.
- *
- * Recommendation: use `.set_as_default_mmr_resource()` to create a single instance of
- * this class per GPU and set them as the default memory resources.
- */
-class GlobalMemoryResource : public DeviceMemoryResource {
```

Comment:
@madsbk I forgot to ask. My feeling is that this may be important for libcudf calls, because not everything uses the provided mr(), but I am not sure where it might be relevant or how much.

Do you have a quick feel/know that this is important, or is it likely unnecessary complexity?
(I think restoring this shouldn't be too bad, since we can skip calling it when there are no GPUs in the system.  But, if we are unsure, also OK to restore it in a followup.)